jaraco/abode/devices/alarm.py

```diff
@@ -23,7 +23,10 @@ def state_from_panel(panel_state, area='1'):
     alarm_state['type'] = 'Alarm'
     alarm_state['type_tag'] = CONST.DEVICE_ALARM
     alarm_state['generic_type'] = CONST.TYPE_ALARM
-    alarm_state['uuid'] = alarm_state.get('mac').replace(':', '').lower()
+    if alarm_state.get('mac'):
+        # On refresh the 'mac' key isn't sent down, just leave the existing UUID if it's
+        # not present
+        alarm_state['uuid'] = alarm_state.get('mac').replace(':', '').lower()
```

Comment:
From what I can tell there's no problem leaving the uuid empty, as it's not consumed anywhere, and as long as the uuid has been set initially it remain in place on refresh.

Reply:
If the UUID isn't being consumed, then we should just remove this functionality.

More importantly, if the mac is only needed in certain cases (initial state, but not refresh), I'd prefer to handle the mac/UUID in that case instead of weaving it's optional presence through the system.

Reply:
I'm not quite sure what the UUID is doing there on Alarm...it looks like it's used to build the device URL for the Light and Camera devices, and not on other types of device, but the property is on the base `Device` class.

Should `_device_uuid` return None by default, and we just ignore it except for those device types that need it?  Seems reasonable.

Then this wouldn't need to try to parse the mac at all and `state_from_panel` doens't need to care if the value is present on the refresh JSON or not.

Reply:
I did some investigation and found that the `uuid` and `device_uuid` and `_device_uuid` are all used by the `.desc()` method, which also expects a UUID to be present for other devices such as the Camera natively (i.e. not derived from a MAC address). I have a different fix incoming.